metron-analytics/metron-profiler-client/src/main/java/org/apache/metron/profiler/client/window/WindowProcessor.java

```diff
@@ -0,0 +1,321 @@
+    Token<?> timeDuration = stack.pop();
+    int duration = ConversionUtils.convert(timeDuration.getValue(), Integer.class);
+    TimeUnit unit = (TimeUnit) timeUnit.getValue();
+    stack.push(new Token<>((int)unit.toMillis(duration), Integer.class));
+  }
+
+  @Override
+  public void exitTimeAmount(org.apache.metron.profiler.client.window.generated.WindowParser.TimeAmountContext ctx) {
+    if(checkForException(ctx)) {
+      return;
+    }
+    if(ctx.getText().length() == 0) {
+      throwable = new IllegalStateException("Unable to parse empty string.");
+      return;
+    }
+    int duration = Integer.parseInt(ctx.getText());
+    stack.push(new Token<>(duration, Integer.class));
+  }
+
+  @Override
+  public void exitTimeUnit(org.apache.metron.profiler.client.window.generated.WindowParser.TimeUnitContext ctx) {
+    checkForException(ctx);
+    switch(normalizeTimeUnit(ctx.getText())) {
+      case "DAY":
+        stack.push(new Token<>(TimeUnit.DAYS, TimeUnit.class));
+        break;
+      case "HOUR":
+        stack.push(new Token<>(TimeUnit.HOURS, TimeUnit.class));
+        break;
+      case "MINUTE":
+        stack.push(new Token<>(TimeUnit.MINUTES, TimeUnit.class));
+        break;
+      case "SECOND":
+        stack.push(new Token<>(TimeUnit.SECONDS, TimeUnit.class));
+        break;
+      default:
+        throw new IllegalStateException("Unsupported time unit: " + ctx.getText()
+                + ".  Supported units are limited to: day, hour, minute, second "
+                + "with any pluralization or capitalization.");
+    }
+  }
+
+  private boolean checkForException(ParserRuleContext ctx) {
+    if(throwable != null)  {
+      return true;
+    }
+    else if(ctx.exception != null) {
+      return true;
+    }
+    return false;
+  }
+
+  private static String normalizeTimeUnit(String s) {
+    String ret = s.toUpperCase().replaceAll("[^A-Z]", "");
+    if(ret.endsWith("S")) {
+      return ret.substring(0, ret.length() - 1);
+    }
+    return ret;
+  }
+
+  public static Window parse(String statement) throws ParseException {
+    if (statement == null || isEmpty(statement.trim())) {
+      return null;
+    }
+    statement = statement.trim();
+    ANTLRInputStream input = new ANTLRInputStream(statement);
+    WindowLexer lexer = new WindowLexer(input);
+    lexer.removeErrorListeners();
+    lexer.addErrorListener(new ErrorListener());
+    TokenStream tokens = new CommonTokenStream(lexer);
+    WindowParser parser = new WindowParser(tokens);
+    WindowProcessor treeBuilder = new WindowProcessor();
+    parser.addParseListener(treeBuilder);
+    parser.removeErrorListeners();
+    parser.addErrorListener(new ErrorListener());
+    parser.window();
+    if(treeBuilder.throwable != null) {
+      throw new ParseException(treeBuilder.throwable.getMessage(), treeBuilder.throwable);
+    }
+    return treeBuilder.getWindow();
+  }
+
+  public static String syntaxTree(String statement) {
```

Comment:
Nobody is using syntaxTree, that's currently only for debugging issues with the grammar going forward for people adapting the language for new use-cases.  Also, @mattf-horton asked for it during the dev discussion.  It was marginally useful during the construction of the grammar to understand how things were being parsed.  I'll document it as such.